src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -104,26 +104,31 @@ public boolean shouldAutoCloseChannel() {
       ScheduledExecutorService getExecutor();
     }
 
-    public Builder() {
+    protected Builder(ConnectionSettings connectionSettings) {
+      this();
+      provideChannelWith(connectionSettings);
```

Comment:
Constructors shouldn't call non-final methods. This one in particular is overridden in derived classes. I didn't notice this problem in the other Builder constructor, but that one needs to be fixed too.